### PR TITLE
[GH-715] Consistent date display

### LINKED ISSUE #715: Bug: Date display is inconsistent
**Steps to reproduce the behavior:**
1. Add a date property to a card and set it
2. Note the display in the Card view (MM/DD/YYYY, e.g. 07/28/2021)
3. Close the card open a Board view
4. Add the date property to the Properties menu, to display in the card preview
5. Note the display in the card preview (MMMM DD, YYYY, e.g. Jul 28, 2021)

**Expected behavior:**
The date display should be consistent between the Card view and Card preview, which should be as follows:
If its the current year: February 25

If its not the current year: February 25, 2020

**Screenshots:**
Date in card view:
Card view date

Date in card preview:
Card preview date

**Platform:**
 - Focalboard v0.8.0 plugin
 - Chrome 91 on macOS 11.4
 
**Additional context:**
Also, the date display on the Card view follows the Date format user setting, but the Card preview does not.

### Recommended change:
* Extend the user date format setting to also include:
  * MMMM DD, YYYY, e.g. July 28, 2021 (make this the default)
  * DD MMMM, YYYY, e.g. 28 July, 2021
* Apply this user setting on all date property displays, including the Card view, card preview, tables, etc.

### PR DESCRIPTION
**Pull request by kamre**
#### Summary
Displaying date is always using `Utils.displayDate` function which uses `MMMM DD` for dates with current year and `MMMM DD, YYYY` for dates with other years.

Here is how it looks:
* for card preview on the board view:
image
* for card dialog view:
image
* inside the row for table view:
image
* editing with day picker in the card dialog:
GH-715-EditableDayPicker

#### Ticket Link
Fixes #715

### CODE REVIEW

**Comment by chenilim** [comment]
@harshilsharma63, could you help review this as sbishel is on PTO. Thanks!

File: webapp/src/utils.test.ts [hunk 1]
```diff
@@ -66,4 +68,21 @@ describe('utils', () => {
             expect(Utils.buildURL('/test')).toBe('base/test')
         })
     })
+
+    describe('display date', () => {
```
Comment (chenilim) [inline webapp/src/utils.test.ts:72]: Thanks for adding a unit test!

**Review by chenilim — APPROVED** [review]
LGTM, thanks @kamre!

**Review by asaadmahmood — APPROVED** [review]

**Review by harshilsharma63 — APPROVED** [review]
LGTM!

**Pull request merged**